### DISCUSSION TITLE
How do build RuleSets?

### DISCUSSION THREAD

**Discussion by code-uri** [evrete/evrete#54] [Q&A] [answered]
As per the documentation a RuleSet is a set of logical rules and their actions can be controlled by  ActivationManager.
I know that Rules can be added to RuleSessions or Knowledge instances. 

Do I need to create multiple instances of knowledge per ruleset? or single knowledge instance is sufficient to add all the the rulesets? if single knowledge is sufficient, then how do control rule activations between two different rulesets who might share same conditions and facts?

**Comment by andbi** [comment, marked as answer]
`Knowledge` extends `RuleSet`, so there can't be multiple rulesets per knowledge.  If you need to split rules into different logical groups inside a single knowledge, you can use the `set("MY-GROUP", "GROUP1")` method to assign them different properties and use these values in your `ActivationManager` as described in the [Conflict resolution](https://evrete.org/docs/conflict-resolution/).

The above mentioned `set` method is somehow still referred to as `property(key, value)`, I will fix it asap, thanks for pointing out the mistake.

**Reply by code-uri** [reply]
Thank you for the quick response.
